Chem master

Dot5511 opened the pull request:
## Short description
Changed the ChemMaster4000's Pill/Patch Output take the solution from the Beaker instead of from the Buffer.

## Why we need to add this
It makes creating Pills/Patches easier for Chemists, encouraging the use of tools at their disposal without having to keep an empty Chemmaster4000. 

## Media (Video/Screenshots)
![image](https://github.com/user-attachments/assets/df71f874-dfe5-4120-a522-d7278ff3cedd)
![image](https://github.com/user-attachments/assets/d977ab2f-95e8-4eef-b80a-b29b7f4680aa)
![image](https://github.com/user-attachments/assets/4f88d5af-f625-4e0b-9be0-bb701dddf557)
![image](https://github.com/user-attachments/assets/7e2fa45b-ff51-4d50-9d8c-eb0757e18e16)

## Checks

- [X] I do not require assistance to complete the PR.
- [X] Before posting/requesting review of a PR, I have verified that the changes work.
- [X] I have added screenshots/videos of the changes, or this PR does not change in-game mechanics.
- [X] I affirm that my changes are licensed under the [Starlight Fork License](https://github.com/ss14Starlight/space-station-14/blob/Starlight/LICENSE-Starlight.TXT) and grant permission for use in this repository under its conditions.

**Changelog**
:cl: Dot5511 and Lokil93
- tweak: Makes the ChemMaster4000's Pill/Patch Output take the solution from the Beaker instead of from the Buffer. 
- add: New text for when the Beaker is Empty/Low and you're trying to make Pills/Patches

Dot5511 committed:
Unstable Container Changes

Changed the Chemmaster 2000 to draw from the Beaker instead of the buffer

Dot5511 committed:
Text & Bugs

Changed the text to say that the Beaker is empty/doesn't have enough solution. Fixed a bug client-wise regarding this.

Rinary1 reviewed (changes requested):
We can't port content from Delta-V, Delta-V it's AGPL-3.0 project.

StarlightHost commented:
Yeah, it’s only a few lines from Delta,
but unfortunately Rinary’s right: since it’s explicitly mentioned that it’s from Delta,
we need direct approval from the author giving us permission to use the change,
with the understanding that we’re not AGPL.

Otherwise, it has to be done from scratch.

Dot5511 committed:
UI Changes

Changed the Client side to be Not Ported from Delta-V, home-made code, enjoy.

Dot5511 committed:
Removed all Delta-V Code

We are now completely free of ANY Delta-V code. Changing Server-Side now.

Dot5511 committed:
Server-Side Changes

Created new code to reflect the change server-side.

Dot5511 commented:
> Yeah, it’s only a few lines from Delta, but unfortunately Rinary’s right: since it’s explicitly mentioned that it’s from Delta, we need direct approval from the author giving us permission to use the change, with the understanding that we’re not AGPL.
> 
> Otherwise, it has to be done from scratch.

We're working on it! So far we did the Client-Side and Server-side changes without including Delta-V stuff but we're looking to also fit the code to make sure that it automatically divides the pills per dosage, like it used to do before with the buffer but instead with the beaker.

Dot5511 committed:
Fixed the Auto-Dosage/Count field for Patches/Pills

Changed the Auto-Dosage/Count fields to refer to the current Beaker volume for values instead of the Chemmaster Buffer.

Dot5511 commented:
Done! All code is now home-made (Using bits and pieces from our own code as well as new variables.) The auto-dosage and auto-count is also now fixed.

StarlightHost commented:
Also add comments like // starlight start and // starlight end around the modified lines
this is so that during an upstream merge, if there’s a conflict with Wizden’s code, it’s clear which parts are ours.

Dot5511 commented:
I'll change it in a bit, thank you very much for the help!

Dot5511 committed:
Created Comments

Included comments for Starlight edits, so as to make it easier to recognize what is Starlight code.

Dot5511 commented:
> Also add comments like // starlight start and // starlight end around the modified lines this is so that during an upstream merge, if there’s a conflict with Wizden’s code, it’s clear which parts are ours.

Sorry for the delay, comments were added!

Happyrobot33 committed:
Update chem-master-component.ftl
